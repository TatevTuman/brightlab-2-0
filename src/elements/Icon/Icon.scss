@import 'static/assets/styles/mixins';

.icon {
  display: flex;

  svg[data-name='like'] {
    path {
    }
  }
}
