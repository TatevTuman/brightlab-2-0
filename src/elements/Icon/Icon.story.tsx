import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'
import Icon, { IconProps } from './Icon'
import { storybook } from '@utils'

interface IconStoryTemplateProps extends IconProps {}

export default {
  title: 'Elements/Icon',
  argTypes: {
    name: {
      control: storybook.controls.icon,
      name: 'Name'
    }
  }
} as Meta

const IconStoryTemplate: Story<IconStoryTemplateProps> = args => {
  return <Icon {...args} />
}

IconStoryTemplate.args = {
  name: 'actions'
}

export const IconStory = IconStoryTemplate.bind({})
IconStory.args = {
  ...IconStoryTemplate.args
}
