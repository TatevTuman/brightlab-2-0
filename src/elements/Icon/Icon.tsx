import React, { CSSProperties } from 'react'
import Times from './icons/times.svg'
import Arrow from './icons/arrow.svg'

import './Icon.scss'

/* ATTENTION! Sync up with @utils/storybook */
export type IconName = 'times' | 'arrow' // | 'your next icon name'

export interface IconProps {
  name: IconName
  className?: string
  style?: CSSProperties
  onClick?(): void
}

const Icon: React.FC<IconProps> = props => {
  const { name } = props

  let icon = null

  switch (name) {
    case 'times': {
      icon = <Times {...props} />
      break
    }
    case 'arrow': {
      icon = <Arrow {...props} />
      break
    }
  }

  if (!icon) return null

  return <i className={'icon'}>{icon}</i>
}

export default Icon
