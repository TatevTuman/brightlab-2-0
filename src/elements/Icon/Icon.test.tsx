import React from 'react'
import { cleanup, render, waitFor } from '@testing-library/react'
import { icons } from '@utils'
import Icon, { IconName } from './Icon'

jest.useFakeTimers()

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Icon', () => {
  it('renders correctly', async () => {
    const { container, rerender } = render(<Icon name={icons[0] as IconName} />)
    await waitFor(() => container)

    icons.forEach(icon => {
      rerender(<Icon name={icon as IconName} />)
      expect(container).toMatchSnapshot()
    })
  })
})
