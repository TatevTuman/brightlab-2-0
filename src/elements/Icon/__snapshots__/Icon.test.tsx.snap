// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<div>
  <i
    class="icon"
  >
    <test-file-stub
      name="times"
    />
  </i>
</div>
`;

exports[`Icon renders correctly 2`] = `
<div>
  <i
    class="icon"
  >
    <test-file-stub
      name="arrow"
    />
  </i>
</div>
`;
